ibm/resource_ibm_satellite_cluster.go

```diff
@@ -0,0 +1,766 @@
+			}
+			params := v1.ClusterUpdateParam{
+				Action:  "update",
+				Force:   true,
+				Version: masterVersion,
+			}
+			err := clusterAPI.Update(clusterID, params, targetEnv)
+			if err != nil {
+				return err
+			}
+			_, err = WaitForSatelliteClusterVersionUpdate(d, meta, targetEnv)
+			if err != nil {
+				return fmt.Errorf(
+					"Error waiting for cluster (%s) version to be updated: %s", d.Id(), err)
+			}
+		}
+		satClient, err := meta.(ClientSession).SatelliteClientSession()
+		if err != nil {
+			return err
+		}
+		workerOpts := satClient.NewGetWorkers1Options(clusterID)
+		workerFields, response, err := satClient.GetWorkers1(workerOpts)
+		if err != nil {
+			return fmt.Errorf("Error retrieving workerFields %s: %s", err, response)
+		}
+
+		getSatClusterOptions := &kubernetesserviceapiv1.GetClusterOptions{
+			Cluster:            &clusterID,
+			XAuthResourceGroup: &resourceGroup,
+		}
+
+		cluster, response, err := satClient.GetCluster(getSatClusterOptions)
+		if err != nil {
+			return fmt.Errorf("Error retrieving cluster %s: %s", clusterID, err)
+		}
+		waitForWorkerUpdate := d.Get("wait_for_worker_update").(bool)
+
+		for _, w := range workerFields {
+			//kubeversion update done if there is a change in Major.Minor version
+			if strings.Split(*w.KubeVersion.Actual, "_")[0] != strings.Split(*cluster.MasterKubeVersion, "_")[0] {
+				params := v1.WorkerUpdateParam{
+					Action: "update",
+				}
+				err = wrkAPI.Update(clusterID, *w.ID, params, targetEnv)
+				if err != nil {
+					d.Set("patch_version", nil)
+					return fmt.Errorf("Error updating worker %s: %s", *w.ID, err)
+				}
+				if waitForWorkerUpdate {
+					_, err = WaitForSatelliteWorkerAvailable(d, meta, targetEnv)
+					if err != nil {
+						return fmt.Errorf(
+							"Error waiting for workers of cluster (%s) to become ready: %s", d.Id(), err)
+					}
+				}
+			}
+		}
+	}
+
+	if d.HasChange("workers") {
+		workerCount := int64(d.Get("workers").(int))
+		resizeOpts := &kubernetesserviceapiv1.V2ResizeWorkerPoolOptions{
+			Cluster:            &clusterID,
+			Workerpool:         &workerPoolName,
+			Size:               &workerCount,
+			XAuthResourceGroup: &resourceGroup,
+		}
+
+		response, err := satClient.V2ResizeWorkerPool(resizeOpts)
+		if err != nil {
+			return fmt.Errorf(
+				"Error updating the workers %d: %s\n%s", workerCount, err, response)
+		}
+	}
+
+	if d.HasChange("labels") {
+		labels := make(map[string]string)
+		if l, ok := d.GetOk("labels"); ok {
+			for k, v := range l.(map[string]interface{}) {
+				labels[k] = v.(string)
+			}
+		}
+
+		wpots := &kubernetesserviceapiv1.V2SetWorkerPoolLabelsOptions{
+			Cluster:            &clusterID,
+			Workerpool:         &workerPoolName,
+			Labels:             labels,
+			XAuthResourceGroup: &resourceGroup,
+		}
+		response, err := satClient.V2SetWorkerPoolLabels(wpots)
+		if err != nil {
+			return fmt.Errorf(
+				"Error updating the labels: %s\n%s", err, response)
+		}
+	}
+
```

Comment:
Add the update support to add or remove additional zones to the cluster

Reply:
Update method will support zones update.